### DISCUSSION TITLE
Mocking nested actions when using jest

### DISCUSSION THREAD

**Discussion by aldibatyr** [mobxjs/mobx-state-tree#1866] [Q&A]
Hello everyone! I am currently trying to implement a test on a store that has an action that calls another action.

Basically the action I am trying to test looks like this:

```
const action = flow(function* (...someArgument) {
yield actionThatIWantToMock/Skip()
...
some logic that I want to test()
}
```

Is there a way to skip the action that is not relevant to the test that I want to do?

I have tried store.actionThatIWantToMock = jest.fn();

also tried `Object.defineProperty` way of setting jest.fn() instead of an actual action :D

Might be doing something wrong, but hit the fence with this one.

Any help/pointers are appreciated!

Thanks a lot!

**Comment by coolsoftwaretyler** [comment]
Hey @aldibatyr - I know this is a bit of an old thread, but I'd love to know: 

1. If you solved your problem, could you post a small code sample for anyone else with a similar issue?
2. If you didn't solve your problem, I'd be happy to help you out if you also post a small sample code with some sample models and Jest test cases. I do quite a bit of testing with Jest and MST and I think I've done what you're looking for here in the past, but I could help you better with a concrete example.

Thank you!
